pkg/purl/purl.go

```diff
@@ -57,6 +57,15 @@ func (p *PackageURL) Package() *ftypes.Package {
 		}
 	}
 
+	// Cocoapods purl has no namespace, but has subpath
+	// https://github.com/package-url/purl-spec/blob/a748c36ad415c8aeffe2b8a4a5d8a50d16d6d85f/PURL-TYPES.rst#cocoapods
+	if p.Type == packageurl.TypeCocoapods && p.Subpath != "" {
+		pkg.Name = strings.Join([]string{
+			p.Name,
+			p.Subpath,
```

Comment:
Isn't the subpath `#Twitter` in the example of `pkg:cocoapods/ShareKit@2.0#Twitter`?

Reply:
I'm not sure I understand what you mean.
`Twitter` is subpath for `pkg:cocoapods/ShareKit@2.0#Twitter`
`NSData+zlib`is subpath for `pkg:cocoapods/GoogleUtilities@7.5.2#NSData+zlib`

Reply:
So, why do you join name and subpath with `/`?

Reply:
`CocoaPods` uses `<ModuleName>/<SubmoduleName>` format.
e.g for `GoogleUtilities`:
```
➜  cat Podfile.lock 
PODS:
  - GoogleUtilities (7.5.2):
    - GoogleUtilities/AppDelegateSwizzler (= 7.5.2)
    - GoogleUtilities/Environment (= 7.5.2)
    - GoogleUtilities/ISASwizzler (= 7.5.2)
    - GoogleUtilities/Logger (= 7.5.2)
    - GoogleUtilities/MethodSwizzler (= 7.5.2)
    - GoogleUtilities/Network (= 7.5.2)
    - "GoogleUtilities/NSData+zlib (= 7.5.2)"
...
```
I join `module` and `submodule` with `/` to reproduce this format.

Reply:
I will add comment for this

Reply:
Got it. Thanks.

Reply:
Added in 775a6c39af9f14214fdf1c08f322b50c12194760

Reply:
Thanks! BTW, is GoogleUtilities a pod name, a module name or a spec name? Is `GoogleUtilities/AppDelegateSwizzler` a subpod, submodule or subspec? "Module might be used in Swift, but not used in CocoaPods. I just want to make sure of the official name.
https://guides.cocoapods.org/syntax/podspec.html

Reply:
We'll go with "module" and fix it later if needed.